samples/java-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/domain/ShoppingCartTestKit.java

```diff
@@ -0,0 +1,53 @@
+package com.example.shoppingcart;
+
+import com.example.shoppingcart.domain.ShoppingCartDomain;
+import com.example.shoppingcart.ShoppingCartApi;
+import com.google.protobuf.Empty;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.Queue;
+import com.akkaserverless.javasdk.testkit.Result;
+
+// Q Would we need ShoppingCartImpl as parameter constructor? 
+public class ShoppingCartTestKit {
+    
+    ShoppingCartDomain.Cart state;
+
+    public ShoppingCartTestKit(ShoppingCartDomain.Cart state){
+        this.state = state;
+    }
+
```

Comment:
This seems more intuitive for the user, to set an initial state from the beginning, so they know where to `get` it and as @octonato suggested they can initiate the test in some complex scenario without the hassle of running the equivalent commands to reach that state.